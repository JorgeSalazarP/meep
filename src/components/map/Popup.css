
.progress-bar {
    background-color: #fff;
    height: 15px;
    padding: 3px;
    width: 60px;
    border-radius: 5px;
    box-shadow: 0 1px 5px #000 inset, 0 1px 0 #444;           
  }
  


  .shine span {
    position: relative;
  }
  
  .shine span::after {
    content: '';
    opacity: 0;
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    background: #fff;
    border-radius: 3px;
    animation: animate-shine 2s ease-out infinite;             
  }
  
  @keyframes animate-shine {
    0% {opacity: 0; width: 0;}
    50% {opacity: .5;}
    100% {opacity: 0; width: 95%;}
  }
  
  
 