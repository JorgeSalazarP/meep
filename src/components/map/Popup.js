import React from 'react';
import T from 'prop-types';
import ProgressBarBattery from './ProgressBarBattery';



const Popup = ({ batteryLevel }) => {

    const [isProgressBar, setIsProgressBar] = React.useState(false);
    const openProgressBarBattery = () => (setIsProgressBar(!isProgressBar));

    
    return ( 
        <React.Fragment>
            <div 
                className="transport-marker"
                onMouseEnter={openProgressBarBattery}
                onMouseOut={openProgressBarBattery}
                     
            >
                <img src="/moto.png" alt="moto"/>

                { isProgressBar
                    &&
                        <div className="progress-bar shine">

                            <ProgressBarBattery
                                batteryLevel={batteryLevel}
                            >
                            </ProgressBarBattery>
                            
                        </div>

                }    
              
            </div> 
            
        </React.Fragment>
    );
}

Popup.propTypes = {
    batteryLevel:T.number
    
};

Popup.defaultProps = {
    batteryLevel: 0,
};
 
export default Popup;