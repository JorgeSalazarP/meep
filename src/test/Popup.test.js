import React from 'react';
import { shallow } from 'enzyme';
import Popup from '../components/map/Popup';

describe('Popup', () => {
    const batteryLevel = 0;
    const render = () => shallow(<Popup batteryLevel={batteryLevel}/>);
    test('should render',()=>{
        const wrapper = render();
        expect(wrapper.exists()).toBe(true);

    });

    test('snapshot testing',()=>{
        const wrapper = render();
        expect(wrapper).toMatchSnapshot();
    });

});
