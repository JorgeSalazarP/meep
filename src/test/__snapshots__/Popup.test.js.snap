// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup snapshot testing 1`] = `
<Fragment>
  <div
    className="transport-marker"
    onMouseEnter={[Function]}
  >
    <img
      alt="moto"
      src="/moto.png"
    />
    <div
      className="progress-bar blue shine"
    >
      <styled.span
        batteryLevel={0}
      />
    </div>
  </div>
</Fragment>
`;
